integration/benchmark/main.go

```diff
@@ -0,0 +1,376 @@
+	flag.Parse()
+
+	chainID := flowsdk.ChainID([]byte(*chainIDStr))
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// parse log level and apply to logger
+	log := zerolog.New(os.Stderr).With().Timestamp().Logger().Output(zerolog.ConsoleWriter{Out: os.Stderr})
+	lvl, err := zerolog.ParseLevel(strings.ToLower(*logLvl))
+	if err != nil {
+		log.Fatal().Err(err).Msg("invalid log level")
+	}
+	log = log.Level(lvl)
+
+	server := metrics.NewServer(log, *metricport, *profilerEnabled)
+	<-server.Ready()
+	loaderMetrics := metrics.NewLoaderCollector()
+
+	if *pushgateway != "" {
+		pusher := push.New(*pushgateway, "loader").Gatherer(prometheus.DefaultGatherer)
+		go func() {
+			t := time.NewTicker(10 * time.Second)
+			defer t.Stop()
+
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-t.C:
+					err := pusher.Push()
+					if err != nil {
+						log.Warn().Err(err).Msg("failed to push metrics to pushgateway")
+					}
+				}
+			}
+		}()
+	}
+
+	accessNodeAddrs := strings.Split(*access, ",")
+
+	cases := parseLoadCases(log, tpsFlag, tpsDurationsFlag)
+
+	addressGen := flowsdk.NewAddressGenerator(chainID)
+	serviceAccountAddress := addressGen.NextAddress()
+	log.Info().Msgf("Service Address: %v", serviceAccountAddress)
+	fungibleTokenAddress := addressGen.NextAddress()
+	log.Info().Msgf("Fungible Token Address: %v", fungibleTokenAddress)
+	flowTokenAddress := addressGen.NextAddress()
+	log.Info().Msgf("Flow Token Address: %v", flowTokenAddress)
+
+	serviceAccountPrivateKeyBytes, err := hex.DecodeString(*serviceAccountPrivateKeyHex)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("error while hex decoding hardcoded root key")
+	}
+
+	ServiceAccountPrivateKey := flow.AccountPrivateKey{
+		SignAlgo: unittest.ServiceAccountPrivateKeySignAlgo,
+		HashAlgo: unittest.ServiceAccountPrivateKeyHashAlgo,
+	}
+	ServiceAccountPrivateKey.PrivateKey, err = crypto.DecodePrivateKey(
+		ServiceAccountPrivateKey.SignAlgo, serviceAccountPrivateKeyBytes)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("error while decoding hardcoded root key bytes")
+	}
+
+	// get the private key string
+	priv := hex.EncodeToString(ServiceAccountPrivateKey.PrivateKey.Encode())
+
+	// sleep in order to ensure the testnet is up and running
+	if *sleep > 0 {
+		log.Info().Msgf("Sleeping for %v before starting benchmark", sleep)
+		time.Sleep(*sleep)
+	}
+
+	loadedAccessAddr := accessNodeAddrs[0]
+	flowClient, err := client.NewClient(loadedAccessAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		log.Fatal().Err(err).Msgf("unable to initialize Flow client")
+	}
+
+	supervisorAccessAddr := accessNodeAddrs[0]
+	if len(accessNodeAddrs) > 1 {
+		supervisorAccessAddr = accessNodeAddrs[1]
+	}
+	supervisorClient, err := client.NewClient(supervisorAccessAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		log.Fatal().Err(err).Msgf("unable to initialize Flow supervisor client")
+	}
+
+	go func() {
```

Comment:
this also looks identical to the other main.  refactor plz

Reply:
I have created a new issue to track this work, separate from this PR.

https://github.com/onflow/flow-go/issues/2965

Reply:
CLI specific code has been removed from the CI version of the loader entrypoint, still keeping the issue I had previously created in order to track further refactor as there is still some shared code that could be further simplified in the future.